include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -701,31 +701,42 @@ struct __deferrable_mode
 {
 };
 
-//A contract for future class: <sycl::event or other event, a value, sycl::buffers..., or __usm_host_or_buffer_storage>
-//Impl details: inheritance (private) instead of aggregation for enabling the empty base optimization.
+// An overload of __wait_and_get_value for 'sycl::buffer'
```

Comment:
Can we modify these function without move them outside?

Reply:
The change also solver a `code coupling` issue.
There was a `code coupling` between following types: __future, 'sycl::buffer', '__result_and_scratch_storage'.

Reply:
But why should move them outside of `class __future` ?